src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.example.bookshop.repository.impl;
+
+import com.example.bookshop.exception.DataProcessingException;
+import com.example.bookshop.model.Book;
+import com.example.bookshop.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
```

Comment:
Think about using annotations that generates constructors. `@AllArgsConstructor` for example.